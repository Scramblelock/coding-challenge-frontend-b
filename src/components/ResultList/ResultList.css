@media only screen and (max-width: 414px) {
  .result-header {
    font-size: 4vw;
  }
}
