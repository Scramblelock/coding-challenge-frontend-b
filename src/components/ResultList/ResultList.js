import React from "react";
import Result from "../Result/Result";
import { useTranslation } from "react-i18next";
import "./ResultList.css";
import moment from "moment";

const formatDate = date => {
  return moment(date).format("MMMM Do YYYY");
};

const ResultList = ({ departures, locations, departureDate }) => {
  const { t } = useTranslation();
  return (
    <div className='result-list'>
      <h2 className='result-header'>
        {t("Here are all the results for")} {formatDate(departureDate)}
      </h2>
      {departures.map((result, index) => {
        const {
          id,
          departure_time,
          arrival_time,
          destination_location_id,
          origin_location_id,
          prices
        } = result;
        const tripPrice = (prices.total / 100).toFixed(2);
        const destinationLocationName = locations.filter(
          location => location.id === destination_location_id
        )[0].name;
        const originLocationName = locations.filter(
          location => location.id === origin_location_id
        )[0].name;
        const departureTime = moment(departure_time).format("LT");
        const arrivalTime = moment(arrival_time).format("LT");

        return (
          <Result
            key={index}
            id={id}
            departureTime={departureTime}
            arrivalTime={arrivalTime}
            tripPrice={tripPrice}
            destinationLocationName={destinationLocationName}
            originLocationName={originLocationName}
          />
        );
      })}
    </div>
  );
};

export default ResultList;
