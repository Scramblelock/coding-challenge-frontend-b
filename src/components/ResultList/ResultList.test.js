import { shallow } from "enzyme";
import React from "react";
import ResultList from "./ResultList";

it("expect to render ResultList component", () => {
  const mockDepartures = [
    {
      id: 1,
      departure_time: "2016-01-14T00:01:00",
      arrival_time: "2016-01-14T07:55:00",
      destination_location_id: 1234,
      origin_location_id: 4321,
      prices: {
        total: 5200
      }
    }
  ];

  const mockLocations = [
    {
      id: 1234,
      city_id: "375dd5879001acbd84a4683dedfb933e",
      name: "Métro Bonaventure Bus Station",
      address: ["997 Rue St-Antoine Ouest", "Montreal, QC H3C 1A6"],
      type: "transit_station",
      lat: 45.4988273060484,
      lon: -73.5644745826722,
      geohash: "f25dvfzcz"
    },
    {
      id: 4321,
      city_id: "5687205723405fhfs024230948",
      name: "Métro Guy Bus Station",
      address: ["997 Rue Guy", "Montreal, QC H3C 1A6"],
      type: "transit_station",
      lat: 45.4988273060484,
      lon: -73.5644745826722,
      geohash: "f25dvfzcz"
    }
  ];

  expect(
    shallow(
      <ResultList departures={mockDepartures} locations={mockLocations} />
    ).debug()
  ).toMatchSnapshot();
});
