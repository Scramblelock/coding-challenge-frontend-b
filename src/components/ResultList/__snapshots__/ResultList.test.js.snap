// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ResultList component 1`] = `
"<div className=\\"result-list\\">
  <h2 className=\\"result-header\\">
    Here are all the results for
     
    August 12th 2019
  </h2>
  <Result id={1} departureTime=\\"12:01 AM\\" arrivalTime=\\"7:55 AM\\" tripPrice=\\"52.00\\" destinationLocationName=\\"Métro Bonaventure Bus Station\\" originLocationName=\\"Métro Guy Bus Station\\" />
</div>"
`;
